emails/models.py

```diff
@@ -338,6 +355,44 @@ def full_address(self):
         return '%s@%s' % (self.address, self.domain_value)
 
 
+def check_user_can_make_another_address(user):
+    user_profile = user.profile_set.first()
+    if (user_profile.at_max_free_aliases and not user_profile.has_premium):
+        hit_limit = f'make more than {settings.MAX_NUM_FREE_ALIASES} aliases'
+        raise CannotMakeAddressException(
+            NOT_PREMIUM_USER_ERR_MSG.format(hit_limit)
+        )
+
+
+def valid_address(address, domain):
+    address_contains_badword = has_bad_words(address)
+    address_is_blocklisted = is_blocklisted(address)
+    address_already_deleted = DeletedAddress.objects.filter(
+        address_hash=address_hash(address, domain=domain)
+    ).count()
+    if (
+        address_already_deleted > 0 or
+        address_contains_badword or
+        address_is_blocklisted
+    ):
+        return False
+    return True
+
+
+def get_domain_from_env_vars_and_profile(user):
+    user_profile = user.profile_set.first()
+    domain = DOMAINS.get('RELAY_FIREFOX_DOMAIN')
+    try:
+        if user_profile.has_premium:
+            domain = DOMAINS.get('MOZMAIL_DOMAIN')
+        else:
+            if settings.TEST_MOZMAIL:
+                domain = DOMAINS.get('MOZMAIL_DOMAIN')
+    except CannotMakeAddressException as e:
+        logger.error(e.message)
```

Comment:
```suggestion
    if user_profile.has_premium or settings.TEST_MOZMAIL:
            domain = DOMAINS.get('MOZMAIL_DOMAIN')
```
Exception will not be thrown here since we are only getting the domain from settings. Simplifying the logic for setting domain as mozmail if the user is a premium user or if the test_mozmail is enabled.